File: R/lpcde_illustration.R
# A simple replication file that illustrates how to use the lpcde package

# load package
library("lpcde")

# software article script
set.seed(42)
n=1000
x_data = matrix(rnorm(2*n, mean=0, sd=1), ncol=2)
#x_data = matrix(rnorm(n, mean=0, sd=1))
y_data = matrix(rnorm(n, mean=0, sd=1))
y_grid = stats::quantile(y_data, seq(from=0.1, to=0.9, by=0.1))

# density estimation
model1 = lpcde::lpcde(x_data=x_data, y_data=y_data, y_grid=y_grid, x=matrix(c(0, 0), ncol=1), bw=0.5)
model1 = lpcde::lpcde(x_data=x_data, y_data=y_data, y_grid=y_grid, x=0, bw=0.5)
summary(model1)

#customizing output
summary(model1, alpha=0.01)

# simple plot
model2 = lpcde::lpcde(x_data=x_data, y_data=y_data, x=0, bw_type = "mse-rot")
plot(model2) + ggplot2::theme(legend.position="none")

#derivative estimation
model3 = lpcde::lpcde(x_data=x_data, y_data=y_data, y_grid=y_grid, x=0, p=3, mu=2)
summary(model3)

#customize plot output
plot(model3, alpha=0.01) + ggplot2::theme(legend.position="none")

# bandwidth selection
y_grid = stats::quantile(y_data, seq(from=0.1, to=0.9, by=0.1))
model4 = lpcde::lpbwcde(y_data=y_data, x_data=x_data, x=0, y_grid = y_grid, bw_type = "mse-rot")
summary(model4)

# IMSE bandwidth selection
model5 = lpcde::lpbwcde(y_data=y_data, x_data=x_data, x=0, y_grid = y_grid, bw_type = "imse-rot")
summary(model5)

File: Python/lpcde_illustration.py
#--------------------------------------------------------------------------------
# LPCDE Package
# Illustration Code
#--------------------------------------------------------------------------------

import numpy as np
import matplotlib

matplotlib.use("Agg")

from lpcde import confint, lpbwcde, lpcde, plot, summary


#----------------------------------------
# Generate data
#----------------------------------------
np.random.seed(42)
n = 500
x_data = np.random.normal(size=n)
y_data = np.random.normal(loc=x_data, scale=1)
y_grid = np.linspace(-1, 1, 5)


#-------------------------------------------------------------------
# lpcde(): estimation with bandwidth 0.5 on provided grid points
#-------------------------------------------------------------------
model1 = lpcde(x_data=x_data, y_data=y_data, y_grid=y_grid, x=0, bw=0.5)
print(summary(model1))
print(confint(model1))


#----------------------------------------
# lpbwcde(): bandwidth selection
#----------------------------------------
model1bw = lpbwcde(y_data=y_data, x_data=x_data, x=0, y_grid=y_grid)
print(summary(model1bw))


#----------------------------------------
# plot(): base matplotlib output
#----------------------------------------
plot(model1)
